Makefile

```diff
@@ -1,58 +1,156 @@
+
+# Build all binaries (alias for build)
+.PHONY: all
+all: build
+
+# Build all binaries
+.PHONY: build
+build: $(TARGETS)
+
+# Build specific binary
 .PHONY: $(TARGETS)
 $(TARGETS):
 	BUILD_PLATFORMS=$(GOOS)/$(GOARCH) hack/build.sh $@
 
-.PHONY: all
-all: $(TARGETS)
-
-# Build image.
-#
-# Args:
-#   GOARCH:      Arch to build.
-#   OUTPUT_TYPE: Destination to save image(docker/registry).
-#
-# Example:
-#   make images
-#   make image-karmada-dashboard-api
-#   make image-karmada-dashboard-api GOARCH=arm64
-IMAGE_TARGET=$(addprefix image-, $(TARGETS))
+###################
+# Docker Images   #
+###################
+
+# Build all images
+.PHONY: images
+images: $(IMAGE_TARGET)
+
+# Build specific image
 .PHONY: $(IMAGE_TARGET)
 $(IMAGE_TARGET):
 	set -e;\
 	target=$$(echo $(subst image-,,$@));\
 	make $$target GOOS=linux;\
 	VERSION=$(VERSION) REGISTRY=$(REGISTRY) BUILD_PLATFORMS=linux/$(GOARCH) hack/docker.sh $$target
 
+###################
+# UI Building     #
+###################
 
+.PHONY: bundle-ui-dashboard
 bundle-ui-dashboard:
 	cd ui && pnpm run dashboard:build
-bin-karmada-dashboard-web:
-	BUILD_PLATFORMS=$(GOOS)/$(GOARCH) hack/build.sh karmada-dashboard-web
-image-karmada-dashboard-web:
-	BUILD_PLATFORMS=linux/$(GOARCH) hack/build.sh karmada-dashboard-web
-	cp -R ui/apps/dashboard/dist _output/bin/linux/$(GOARCH)/dist
-	DOCKER_FILE="build-web.Dockerfile" VERSION=$(VERSION) REGISTRY=$(REGISTRY) BUILD_PLATFORMS=linux/$(GOARCH) hack/docker.sh karmada-dashboard-web
 
+###################
+# Dependencies    #
+###################
+
+# Install Go dependencies
+.PHONY: install-deps
+install-deps:
+	go mod tidy
+	go mod verify
+
+# Install UI dependencies
+.PHONY: install-ui-deps
+install-ui-deps:
+	cd ui && pnpm install
+
+# Install all dependencies
+.PHONY: install
+install: install-deps install-ui-deps
+
+###################
+# Development     #
+###################
+
+# Run both API and Web server
+.PHONY: run
+run:
+ifndef KUBECONFIG
+	$(error KUBECONFIG is required. Please specify the path to karmada kubeconfig)
+endif
+	@echo "Starting API server..."
+	@_output/bin/$(GOOS)/$(GOARCH)/karmada-dashboard-api \
+		--karmada-kubeconfig=$(KUBECONFIG) \
+		--karmada-context=$(KARMADA_CTX) \
+		--kubeconfig=$(KUBECONFIG) \
+		--context=$(HOST_CTX) \
+		--insecure-port=$(API_PORT) & echo $$! > .api.pid
+	@echo "Starting Web server..."
+	@cd ui && VITE_API_HOST=$(API_HOST) pnpm run dashboard:dev || (kill `cat .api.pid` && rm .api.pid)
+	@if [ -f .api.pid ]; then kill `cat .api.pid` && rm .api.pid; fi
+
+# Run API server only
+.PHONY: run-api
+run-api:
+ifndef KUBECONFIG
+	$(error KUBECONFIG is required. Please specify the path to karmada kubeconfig)
+endif
+	_output/bin/$(GOOS)/$(GOARCH)/karmada-dashboard-api \
+		--karmada-kubeconfig=$(KUBECONFIG) \
+		--karmada-context=$(KARMADA_CTX) \
+		--kubeconfig=$(KUBECONFIG) \
+		--context=$(HOST_CTX) \
+		--insecure-port=$(API_PORT)
+
+# Run Web server only
+.PHONY: run-web
+run-web:
```

Comment:
same as `run-api`, it should depend on build task

Reply:
added, and also depend install-ui-deps?